src/coreclr/nativeaot/Runtime/amd64/CachedInterfaceDispatchAot.S

```diff
@@ -0,0 +1,27 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+.intel_syntax noprefix
+#include <AsmOffsets.inc>         // generated by the build from AsmOffsets.cpp
+#include <unixasmmacros.inc>
+
+// trick to avoid PLT relocation at runtime which corrupts registers
+#define REL_C_FUNC(name) C_FUNC(name)@gotpcrel
+
+// Stub dispatch routine for dispatch to a vtable slot
+LEAF_ENTRY RhpVTableOffsetDispatch, _TEXT
+        int 3
+        // UNIXTODO: Implement this function
+    ALTERNATE_ENTRY RhpVTableOffsetDispatchAVLocation
+        int 3
+LEAF_END RhpVTableOffsetDispatch, _TEXT
+
+// Cache miss case, call the runtime to resolve the target and update the cache.
+// Use universal transition helper to allow an exception to flow out of resolution
+LEAF_ENTRY RhpInterfaceDispatchSlow, _TEXT
+        // r11 contains indirection cell address already, so it will naturally be passed to RhpCidResolve
+        // the universal transition thunk as an argument to RhpCidResolve
+        mov r10, [rip + REL_C_FUNC(RhpCidResolve)]
```

Comment:
clr-abi.md says we preserve r10 but it's being overwritten here, is r10 preserved elsewhere?